velox/exec/HashJoinBridge.h

```diff
@@ -32,13 +32,35 @@ class HashJoinBridgeTestHelper;
 /// the same name.
 class HashJoinBridge : public JoinBridge {
  public:
+  struct SpillResult {
+    Spiller* spiller{nullptr};
+    const std::exception_ptr error{nullptr};
+
+    explicit SpillResult(std::exception_ptr _error) : error(_error) {}
+    explicit SpillResult(Spiller* _spiller) : spiller(_spiller) {}
+  };
+
   void start() override;
 
   /// Invoked by HashBuild operator ctor to add to this bridge by incrementing
   /// 'numBuilders_'. The latter is used to split the spill partition data among
   /// HashBuild operators to parallelize the restoring operation.
   void addBuilder();
 
+  /// Invoked to spill 'table' and returns spilled partitions. This method
+  /// should only be invoked when the 'table' is a ready-to-use one, meaning it
+  /// should not be one in the middle of building. Hence it is normally invoked
+  /// by probe side.
+  SpillPartitionSet spillTable(
```

Comment:
Let's move spillTable out of HashJoinBridge?

Reply:
We cannot, spillTable depends on the member of bridge when it comes to the scenario where we spill directly from join bridge.